syndiffix/synthesizer.py

```diff
@@ -79,6 +83,12 @@ def __init__(
             apply_convertors(self.column_convertors, raw_data),
         )
 
+        if not clustering:
+            if target_column is not None:
```

Comment:
Both of these conditions test for `None`, but they do it in a different way.
Also, I think validation should also be done here instead of separately.
```py
if target_column:
    if clustering:
        raise ValueError("Cannot specify both target_column and clustering parameters.")
    clustering = MlClustering(target_column=target_column)
 elif not clustering:
        clustering = DefaultClustering()
```

Reply:
Good point. Though we should move this up before we do any time-consuming work, otherwise you'd have to wait a couple of seconds just to get the value error.

Reply:
> Both of these conditions test for None, but they do it in a different way.

We allow for `target_column = ColumnId(0)` which is falsy.